wow/macros/src/main/scala/wow/common/database/ExtraFieldsSQLInterpolationMacro.scala

```diff
@@ -3,36 +3,96 @@ package wow.common.database
 import scalikejdbc.interpolation.SQLSyntax
 
 import scala.annotation.StaticAnnotation
+import scala.collection.mutable
 import scala.reflect.macros._
 
 //noinspection SpellCheckingInspection
 case class databasefields(s: String*) extends StaticAnnotation
 
+//noinspection SpellCheckingInspection
+class databasecomponent extends StaticAnnotation
+
+//noinspection SpellCheckingInspection
+class databaseprefix extends StaticAnnotation
+
 /**
   * Macros for dynamic fields validation
   */
 object ExtraFieldsSQLInterpolationMacro {
   def selectDynamic[E: c.WeakTypeTag](c: blackbox.Context)(name: c.Expr[String]): c.Expr[SQLSyntax] = {
     import c.universe._
 
+    /**
+      * Extracts name of symbol
+      * @param s symbol
+      * @return name
+      */
     def getName(s: c.universe.Symbol) = s.name.encodedName.toString.trim
 
+    /**
+      * Is symbol annotated with databasecomponent
+      * @param symbol symbol
+      * @return true if annotated, false otherwise
+      */
+    def isComponentAnnotated(symbol: c.universe.Symbol) = symbol.annotations.exists(a => a.tree.tpe == typeOf[databasecomponent])
+
+    def getPrefix(symbol: c.universe.Symbol) = {
+      symbol.annotations.find(a => a.tree.tpe == typeOf[databaseprefix]) match {
+        case Some(a) =>
+          a.tree.children.tail.collectFirst({ case Literal(Constant(field: String)) => field }).get
+        case None =>
+          ""
```

Comment:
No because that function returns a string and we want to be able to concatenate that with another string. "" is therefore a valid value (e.g. it's not meant to indicate there is anything wrong, just that it's not specified by default, but that is correct behavior).